src/post/post.entity.ts

```diff
@@ -199,6 +217,26 @@ export class Post extends BaseEntity {
         this.handleDeleted();
     }
 
+    update(account: Account, params: PostUpdateParams) {
+        if (account.uuid !== this.author.uuid) {
+            throw new Error(
+                `Account ${account.uuid} cannot update Post ${this.uuid}`,
+            );
+        }
+        this._updateDirty = true;
+
+        // TODO: Clean up the any type
+        // biome-ignore lint/suspicious/noExplicitAny: Legacy code needs proper typing
+        const self = this as any;
```

Comment:
We shouldn't be introducing new usage of `any` here - this breaks type checking, if `PostUpdateParams` is out of sync with `Post` then we won't catch it!